modules/core/shared/src/main/scala/lucuma/core/model/sequence/SequenceDigest.scala

```diff
@@ -35,22 +35,14 @@ case class SequenceDigest(
   executionState: ExecutionState
 ):
 
-  def add(o: ObserveClass): SequenceDigest =
-    SequenceDigest.observeClass.modify(_ |+| o)(this)
-
-  def add(p: CategorizedTime): SequenceDigest =
-    SequenceDigest.timeEstimate.modify(_ |+| p)(this)
-
-  def add(o: Offset): SequenceDigest =
-    SequenceDigest.offsets.modify(_ + o)(this)
-
-  def incrementAtomCount: Option[SequenceDigest] =
-    NonNegInt
-      .from(atomCount.value + 1)
-      .toOption
-      .map { ac =>
-        SequenceDigest.atomCount.replace(ac)(this)
-      }
+  def add[D](a: Atom[D]): SequenceDigest =
+    SequenceDigest(
+      observeClass   = observeClass |+| a.observeClass,
+      timeEstimate   = timeEstimate |+| a.timeEstimate,
+      offsets        = a.steps.toList.foldLeft(offsets)((os, s) => os + s.telescopeConfig.offset),
+      atomCount      = NonNegInt.unsafeFrom(atomCount.value + 1),
+      executionState = executionState
+    )
```

Comment:
This is more or less unrelated but fell out of an aborted previous version and strikes me as better than the individual `add`s that the ODB was previously using.